codegen/src/main/java/software/amazon/awssdk/codegen/poet/builder/BaseClientBuilderClass.java

```diff
@@ -288,16 +288,6 @@ private MethodSpec finalizeServiceConfigurationMethod() {
         builder.addCode("interceptors = $T.mergeLists(interceptors, config.option($T.EXECUTION_INTERCEPTORS));\n",
                         CollectionUtils.class, SdkClientOption.class);
 
-        if (model.getMetadata().isQueryProtocol()) {
-            TypeName listType = ParameterizedTypeName.get(List.class, ExecutionInterceptor.class);
-            builder.addStatement("$T protocolInterceptors = $T.singletonList(new $T())",
-                                 listType,
-                                 Collections.class,
-                                 QueryParametersToBodyInterceptor.class);
-            builder.addStatement("interceptors = $T.mergeLists(interceptors, protocolInterceptors)",
-                                 CollectionUtils.class);
-        }
```

Comment:
Not sure I understand the need for the other changes. Any reason we can't just change the ordering of this statement?

Reply:
In this `finalizeServiceConfiguration()` the service-specific interceptors are added to the `SdkClientConfiguration` as `SdkClientOption.EXECUTION_INTERCEPTORS` which are then merged with the global interceptors in [SdkDefaultClientBuilder](https://github.com/aws/aws-sdk-java-v2/blob/master/core/sdk-core/src/main/java/software/amazon/awssdk/core/client/builder/SdkDefaultClientBuilder.java#L435). So changing the ordering here would only move the interceptor to the front of the service-specific interceptors; it would still come after the global interceptors. 

Alternatively, we could create another SdkClientOption here and merge it in SdkDefaultClientBuilder, but to keep cross-module dependencies down Matt proposed we add it in the `SyncClientBuilderClass` / `AsyncClientBuilderClass`